cpp/lib/src/IssnLookup.cc

```diff
@@ -0,0 +1,120 @@
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <iostream>
+#include <nlohmann/json.hpp>
+#include "Downloader.h"
+#include "IssnLookup.h"
+
+
+namespace IssnLookup {
+
+
+const unsigned int TIMEOUT_IN_SECONDS(15);
+
+
+void ExtractingData(ISSNInfo * const issn_info, const nlohmann::json &issn_info_json, const std::string issn) {
+    const std::string issn_uri("resource/ISSN/" + issn), issn_title_uri("resource/ISSN/" + issn + "#KeyTitle");
+    if (issn_info_json.at("@graph").is_structured()) {
+        for (auto ar : issn_info_json.at("@graph")) {
+            if (ar.is_structured()) {
+                if (ar.at("@id") == issn_uri) {
+                    for (auto &[key, val] : ar.items()) {
+                        if (key == "mainTitle")
+                            issn_info->main_title_ = val;
+                        if (key == "format")
+                            issn_info->format_ = val;
+                        if (key == "identifier")
+                            issn_info->identifier_ = val;
+                        if (key == "type")
+                            issn_info->type_ = val;
+                        if (key == "issn")
+                            issn_info->issn_ = val;
+                        if (key == "isPartOf")
+                            issn_info->is_part_of_ = val;
+                        if (key == "publication")
+                            issn_info->publication_ = val;
+                        if (key == "url")
+                            issn_info->url_ = val;
+                        if (key == "name") {
+                            if (val.is_structured()) {
+                                for (auto val_item : val)
+                                    issn_info->names_.emplace_back(val_item);
+                            } else {
+                                issn_info->names_.emplace_back(val);
+                            }
+                        }
+                    }
+                }
+                if (ar.at("@id") == issn_title_uri) {
+                    for (auto &[key, val] : ar.items()) {
+                        if (key == "value")
+                            issn_info->title_ = val;
+                    }
+                }
+            }
+        }
+    }
+}
+
+bool GetISSNInfo(const std::string &issn, ISSNInfo * const issn_info) {
+    const std::string issn_url("https://portal.issn.org/resource/ISSN/" + issn + "?format=json");
+
+    Downloader downloader(issn_url, Downloader::Params(), TIMEOUT_IN_SECONDS * 1000);
```

Comment:
As far as i know, you can remove the timeout & params arguments from the function call so the default values will be taken (also no need to override the default timeout, so you can also remove the constant definition entirely.)